### DISCUSSION TITLE
Wasm cannot support Chinese fonts

### DISCUSSION THREAD

**Discussion by xsteel2003** [AvaloniaUI/Avalonia#9576] [Q&A]
private void MainView_Loaded(object? sender, Avalonia.Interactivity.RoutedEventArgs e)
        {
            Text1.FontFamily = new Avalonia.Media.FontFamily(new System.Uri(@"/Assets/Fonts/Fontquan-XinYiGuanHeiTi.ttf", System.UriKind.Relative), "Fontquan-XinYiGuanHeiTi");
        }

Hello everyone, the above code, displays the correct font on Windows. But on the web, boxes are displayed. what went wrong?

**Comment by xsteel2003** [comment]
[Fontquan-XinYiGuanHeiTi.zip](https://github.com/AvaloniaUI/Avalonia/files/10126578/Fontquan-XinYiGuanHeiTi.zip)
Here is the font file.

**Reply by timunie** [reply]
Which Avalonia version do you use?

**Reply by xsteel2003** [reply]
VS2012 template is 0.10.18 , nuget is 11.0.0-preview4

**Reply by timunie** [reply]
Can you try with latest nightly build again? I see many PRs about text processing, so worth a try. 

https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed

**Comment by maxkatz6** [comment]
I don't think this method accepts relative baseUri: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Media/FontFamily.cs#L24-L50

**Reply by xsteel2003** [reply]
i tried
 `Text1.FontFamily = new Avalonia.Media.FontFamily(new System.Uri("avares://AvaloniaApplication10/Assets/Fonts/Fontquan-XinYiGuanHeiTi.ttf", System.UriKind.Absolute), "Fontquan-XinYiGuanHeiTi");` 
and 
` Text1.FontFamily = new Avalonia.Media.FontFamily(new System.Uri("resm:/Assets/Fonts/Fontquan-XinYiGuanHeiTi.ttf", System.UriKind.Absolute), "Fontquan-XinYiGuanHeiTi");`
But it also only works under windows. Font effects cannot be displayed under both Web and Android.

**Reply by maxkatz6** [reply]
Correct font name could be "Fontquan" (as a second argument of FontFamily). 
IIRC font lookup depends on the platform.

**Comment by xsteel2003** [comment]
![image](https://user-images.githubusercontent.com/16742748/204994439-56d4f259-a691-4487-a229-c4d07cdc48d1.png)


![image](https://user-images.githubusercontent.com/16742748/204994260-40a0edd9-b828-492f-a9fa-042367ec0975.png)

**Comment by rabbitism** [comment]
This is what I configured and it works.
```csharp
private static void Main(string[] args) => BuildAvaloniaApp()
        .UseReactiveUI()
        .With(new FontManagerOptions
        {
            FontFallbacks = new[]
            {
                new FontFallback
                {
                    FontFamily = new FontFamily("avares://WebTest/Assets/Fontquan-XinYiGuanHeiTi.ttf#Fontquan")
                }
            }
        })
        .SetupBrowserApp("out");
```
![image](https://user-images.githubusercontent.com/14807942/205684352-1156004e-d8cb-4cd8-867c-30ab66f84d47.png)
![image](https://user-images.githubusercontent.com/14807942/205684464-f55c6584-bd3d-48e4-9650-e6e170c5f023.png)
Reference: https://github.com/AvaloniaUI/Avalonia/issues/8938
